apps/site/lib/site_web/plugs/retirement_message.ex

```diff
@@ -0,0 +1,31 @@
+defmodule SiteWeb.Plugs.RetirementMessage do
+  @moduledoc """
+  Assigns @retirement_message based on a `retirement_message` parameter built beforehand.
+  """
+  @behaviour Plug
+  import Plug.Conn, only: [assign: 3]
+  import Phoenix.HTML.Tag, only: [content_tag: 3]
+  import Phoenix.HTML, only: [safe_to_string: 1]
+  import SiteWeb.Router.Helpers, only: [line_path: 3]
+
+  defp message_header(), do: "We'll be retiring this tool on [Date]."
+
+  defp message_body(conn) do
+    path = line_path(conn, :show, conn.assigns[:route].id)
+    tag = content_tag(:a, "Schedule Finder", href: path)
+
+    link_to_schedule_finder = safe_to_string(tag)
+
+    "All the information provided here, including scheduled trips and realtime departure and arrival information, can be found in our #{
+      link_to_schedule_finder
+    }."
+  end
+
+  @impl true
+  def init(_), do: %{header: "", body: ""}
```

Comment:
Just noting here that since the second argument in the call function isn't used, this map that's returned here doesn't see the light of day. 🤷‍♀️ I think it's ok to not return anything here.